### ISSUE TITLE
[CLI] Keras WandbCallback logs summary of last epoch instead of best

### ISSUE THREAD

**Issue by bzamecnik** [wandb/client#2756]
**Description**

WandbCallback docs:
> `WandbCallback` will set summary metrics for the run associated with the "best" training
step, where "best" is defined by the `monitor` and `mode` attribues [sic, typo].

But in practice the summary metrics are values from the last epoch, except for extra `best_<monitored_metric>`.

**Wandb features**

`wandb.integration.keras.keras.WandbCallback`

**How to reproduce**
1. Link to a reproducible script we can run to see the bug
2. Describe what we have to do in order to trigger the issue
3. Add a zip file with the run folder 

```
model.fit(X_train,
            y_train,
            validation_data=(X_test, y_test),
            callbacks=[WandbCallback(monitor="val_loss", mode="min")]
        )
```

- model produces multiple other metrics
- model is trained in several epochs and the best one (where loss is minimal) is not the last one
- wandb summary contains:
  - best_val_loss - correct value from best epoch
  - other metrics - value from last epoch (in contrast to the docs)

**Environment**
- `wandb==0.12.4`
- OS: Ubuntu 18.04
- Environment: virtualend,
- Python Version: 3.6.8

**Comment by sydholl** [comment]
We are closing this due to inactivity, please comment to reopen.
